rust_src/src/symbols.rs

```diff
@@ -404,5 +403,40 @@ pub fn symbol_value(symbol: LispSymbolRef) -> LispObject {
     }
     val
 }
+/// Non-nil if VARIABLE has a local binding in buffer BUFFER.
+/// BUFFER defaults to the current buffer.
+#[lisp_fn(min = "1")]
+pub fn local_variable_p(mut symbol: LispSymbolRef, buffer: LispBufferOrCurrent) -> bool {
+    let buf: LispBufferRef = buffer.into();
+
+    symbol = symbol.get_indirect_variable();
+
+    match symbol.get_redirect() {
+        symbol_redirect::SYMBOL_PLAINVAL => false,
+        symbol_redirect::SYMBOL_LOCALIZED => {
+            let blv = unsafe { symbol.get_blv() };
+            if blv.where_.eq(buf) {
+                blv.found()
+            } else {
+                let variable: LispObject = symbol.into();
+                buf.local_vars_iter().any(|local_var| {
+                    let (car, _) = local_var.into();
+                    variable.eq(car)
+                })
+            }
+        }
+        symbol_redirect::SYMBOL_FORWARDED => unsafe {
+            let contents = symbol.get_fwd();
+            if !BUFFER_OBJFWDP(contents) {
+                return false;
+            }
+
+            let offset = (*contents).u_buffer_objfwd.offset;
+            let idx = per_buffer_idx(offset.get_byte_offset() as isize);
+            idx == -1 || buf.value_p(idx as isize)
+        },
+        _ => unreachable!(),
```

Comment:
With the PR I just pushed this can be rewritten like this:

        symbol_redirect::SYMBOL_FORWARDED => unsafe {
            let contents = symbol.get_fwd();
            match as_buffer_objfwd(contents) {
                Some(buffer_objfwd) => {
                    let idx = per_buffer_idx_from_field_offset(buffer_objfwd.offset);
                    idx == -1 || buf.value_p(idx as isize)
                }    
                None => false
            }
        },